rustls-platform-verifier/src/verification/others.rs

```diff
@@ -54,11 +54,17 @@ impl Verifier {
     /// WebPKI, using root certificates provided by the platform and augmented by
     /// the provided extra root certificates.
     pub fn new_with_extra_roots(
-        roots: impl IntoIterator<Item = pki_types::TrustAnchor<'static>>,
+        roots: impl IntoIterator<Item = pki_types::CertificateDer<'static>>,
```

Comment:
I think the tricky part here would be that the conversion from `CertificateDer` to `TrustAnchor` is a non-trivial amount of DER parsing that currently lives in webpki. So far, pki-types contains very little logic, and my intuition is that it might be better to keep it that way?

Reply:
That seems like a reasonable middleground. With that this could be a bit cleaner and push the bulkiness of roots further away from this crate (and probably others)

Reply:
@ctz @cpu thoughts?

Reply:
I think I would prefer to allow composition of verifiers at the rustls level, rather than relax the type safety of `TrustAnchor`.

But, then, I think that's because I'm not 100% sold on the value of giving extra roots to the platform verifier (I previously thought it was a method for affixing the behaviour for testing purposes, but I guess not?)